Automatic start at boot not working with Wayland

On reboot, KlipperScreen does not successfully start until I log in by SSH. `syslog` reports:

```
systemd[1]: KlipperScreen.service: Scheduled restart job, restart counter is at 1.
systemd[1]: Started KlipperScreen.service - KlipperScreen.
KlipperScreen-start.sh[970]: Running KlipperScreen on Cage
seatd[665]: 00:00:03.156 [INFO] [seatd/server.c:145] New client connected (pid: 974, uid: 1000, gid: 1003)
seatd[665]: 00:00:03.156 [INFO] [seatd/seat.c:170] Added client 1 to seat0
seatd[665]: 00:00:03.156 [INFO] [seatd/seat.c:480] Opened client 1 on seat0
KlipperScreen-start.sh[974]: 00:00:00.007 [ERROR] [EGL] command: eglQueryDeviceStringEXT, error: EGL_BAD_PARAMETER (0x300c), message: "eglQueryDeviceStringEXT"
KlipperScreen-start.sh[974]: 00:00:00.007 [ERROR] [EGL] command: eglQueryDeviceStringEXT, error: EGL_BAD_PARAMETER (0x300c), message: "eglQueryDeviceStringEXT"
KlipperScreen-start.sh[974]: 00:00:00.044 [ERROR] [../cage.c:541] Unable to open Wayland socket: Invalid argument
seatd[665]: 00:00:03.201 [INFO] [seatd/seat.c:418] No clients on seat0 to activate
seatd[665]: 00:00:03.201 [INFO] [seatd/seat.c:524] Closed client 1 on seat0
seatd[665]: 00:00:03.201 [INFO] [seatd/seat.c:192] Removed client 1 from seat0
seatd[665]: 00:00:03.201 [INFO] [seatd/client.c:471] Client disconnected
systemd[1]: KlipperScreen.service: Main process exited, code=exited, status=1/FAILURE
systemd[1]: KlipperScreen.service: Failed with result 'exit-code'.
```

When I ssh in, `/var/run/user/1000/wayland-0` is created by my SSH session, which then allows the KlipperScreen service to start successfully.

The fix for me is to to add a `PAMName` line to the `[Service]` block of the unit file:

```
diff --git a/scripts/KlipperScreen.service b/scripts/KlipperScreen.service
index bf209488..c9a30a5a 100644
--- a/scripts/KlipperScreen.service
+++ b/scripts/KlipperScreen.service
@@ -25,6 +25,8 @@ ExecStart="KS_DIR/scripts/KlipperScreen-start.sh"
 # 'who'. This is needed since we replace (a)getty.
 UtmpIdentifier=tty7
 UtmpMode=user
+# Create a PAM session to run pam-systemd, which creates /var/run/$UID
+PAMName=%u
 # A virtual terminal is needed.
 TTYPath=/dev/tty7
 TTYReset=yes
```

I've tested this change when running under X, and it also works fine -- the only difference I can tell is that a systemd login session is created:

```
❯ sudo loginctl
SESSION  UID USER    SEAT  TTY  STATE  IDLE SINCE
      2 1000 russell -     -    active no   -
     c1 1000 russell seat0 tty7 online no   -
```

(second entry, on tty7).

I'm a bit hesitant to put up a pull request to just add this, as it seems suspicious to me that nobody else has run into this. Maybe not many people run on Wayland? I'm also wary that the particular change I've suggested will change how the X config works, and I don't know systemd well enough to know if it might have any knock-on impacts. I guess it could be another thing that the install script substitutes during Wayland install?

In case it makes any difference, I'm running Ubuntu Server 24.04.

--- a/scripts/KlipperScreen-install.sh
+++ b/scripts/KlipperScreen-install.sh
@@ -37,7 +37,7 @@ install_graphical_backend()
       echo_text ""
       echo_text "Choose graphical backend"
       echo_ok "Default is Xserver"
-      echo_text "Wayland is EXPERIMENTAL needs kms/drm drivers doesn't support DPMS and may need autologin"
+      echo_text "Wayland is EXPERIMENTAL, needs kms/drm drivers, and doesn't support DPMS"
       echo_text ""
       echo "Press enter for default (Xserver)"
       read -r -e -p "Backend Xserver or Wayland (cage)? [X/w]" BACKEND

--- a/scripts/KlipperScreen.service
+++ b/scripts/KlipperScreen.service
@@ -25,6 +25,8 @@ ExecStart="KS_DIR/scripts/KlipperScreen-start.sh"
 # 'who'. This is needed since we replace (a)getty.
 UtmpIdentifier=tty7
 UtmpMode=user
+# Create a PAM session to run pam-systemd, which creates /var/run/$UID
+PAMName=%u
 # A virtual terminal is needed.
 TTYPath=/dev/tty7
 TTYReset=yes

